### DISCUSSION TITLE
Third Person Follow Mode Becomes First Person with Jolt Physics

### DISCUSSION THREAD

**Discussion by ElektroFox** [ramokz/phantom-camera#239] [Question / Help Wanted] [answered]
With JoltPhysics freshly installed, upon restart and testing my scene, the Third Person PCam that was working just fine now zooms into the character capsule as if it were assuming the position of a first person camera. I cannot find anything in the documentation for this. Any idea on how to fix this? Is it a bug?

EDIT: I reset to Godot's default physics, then back to Jolt (restarting the editor for each) and all of a sudden it is working. I changed nothing except for the physics engine. Quite odd. I would provide footage, but I did not have the ability to recreate the bug while attempting to re-record for expediency's sake. If I am able to replicate the issue, I will update this post with video.

EDIT 2: After modifying some of the spring settings in the PhantomCamera node on my player, the bug has returned. Video below.

https://github.com/ramokz/phantom-camera/assets/13038096/d0c1bdd2-fef1-478f-afd7-3cb2621843e8

**Comment by ramokz** [comment]
My guess would be that it's related to the `Collision Mask`.

If you disable all the layers, in your case it's just layer 1, does the issue still persist?

**Reply by ramokz** [reply]
As an aside, you also may want to avoid nesting both `Camera3D` and `PCam` inside its follow target.

Mainly because it means that every transform change for the target affects the camera — be it positional or rotational.
It also disables some core features behind the system, such as the `follow damping`.

**Reply by ElektroFox** [reply]
I've moved the camera and phantom camera out of the player scene, and no difference between behavior seems apparent to me. Disabling the Spring Arm collision masks entirely solves the issue, but this isn't an ideal solution. If I switch the Spring Arm to collision layer 2, the problem is resolved as well, but I'm unsure why exactly that is the case, so I'm hesitant to treat that as a real solution yet.

**Reply by ramokz** [reply, marked as answer]
That is an expected result.

What is happening is that the collision mask of the `SpringArm3D` is colliding with the player collider (`PlayerBody`). The `SpringArm3D` is positioned at the centre of its target, so the line between it and the `PCam` is blocked by player collider - hence the camera result.

So the solve is to just ensure the player collider and the `PCam` (again, internally the `SpringArm3D`) aren't on the same collision layer.
